pkg/common/cns-lib/vsphere/virtualmachine.go

```diff
@@ -439,16 +439,14 @@ func (vm *VirtualMachine) GetTopologyLabels(ctx context.Context, tagManager *tag
 			val, exists := topologyCategories[category.Name]
 			log.Debugf("Found category %q for object %+v with tag: %q ", val, obj.Self, tag.Name)
 			if exists {
-				// Update the value if it doesn't already exist.
-				if val == "" {
-					log.Infof("Found category: %s for object %v with tag: %s",
-						category.Name, obj.Self, tag.Name)
-					topologyCategories[category.Name] = tag.Name
-				} else {
-					// Error out on duplicate values for the same category.
-					return logger.LogNewErrorf(log, "duplicate values detected for category %s as %q and %q",
-						category.Name, val, tag.Name)
-				}
+				// If the category already exists, update the value.
```

Comment:
Could you maintain the `if..else..` block as before and log a warning message in the else block stating that duplicate values have been found for the same category. We do not want to encourage users to use this kind of configuration as it needs special handling everytime we develop features around topology.

Reply:
Yes that makes sense, updated. Thanks!
```
{"level":"warn","time":"2024-03-27T23:26:55.738243634Z","caller":"vsphere/virtualmachine.go:451","msg":"duplicate values detected for category openshift-zone as \"zone-C\" and \"zone-C\""}
```